### PR TITLE
TextControl: Fix remaining 40px size violations

### PR DESCRIPTION
**Pull request by mirka**
Part of #63871

## What?

Fixes the remaining 40px size violations for TextControl.

## Testing Instructions

See inline comments.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -319,6 +319,7 @@ module.exports = {
 						'FontSizePicker',
 						'NumberControl',
 						'RangeControl',
+						'TextControl',
```
Comment (mirka) [inline .eslintrc.js:322]: Moved to the stricter rule that doesn't allow `__next40pxDefaultSize={ false }`, now that we're done migrating.

File: packages/block-editor/src/components/date-format-picker/index.js [hunk 2]
```diff
@@ -150,8 +150,7 @@ function NonDefaultControls( { format, onChange } ) {
 			/>
 			{ isCustom && (
 				<TextControl
-					// TODO: Switch to `true` (40px size) if possible
-					__next40pxDefaultSize={ false }
+					__next40pxDefaultSize
```
Comment (mirka) [inline packages/block-editor/src/components/date-format-picker/index.js:153]: | Before | After |
|--------|-------|
|DateFormatPicker, before|DateFormatPicker, after|

File: packages/block-editor/src/components/url-popover/image-url-input-ui.js [hunk 3]
```diff
@@ -227,16 +227,14 @@ const ImageURLInputUI = ( {
 				checked={ linkTarget === '_blank' }
 			/>
 			<TextControl
-				// TODO: Switch to `true` (40px size) if possible
-				__next40pxDefaultSize={ false }
+				__next40pxDefaultSize
```
Comment (mirka) [inline packages/block-editor/src/components/url-popover/image-url-input-ui.js:230]: Image block ▸ "Link" in block toolbar ▸ Down caret ("Link settings") dropdown in the popover.

| Before | After |
|--------|-------|
|Image block URL popover, before|Image block URL popover, after|

File: packages/block-library/src/form-input/edit.js [hunk 4]
```diff
@@ -63,8 +63,7 @@ function InputFieldBlock( { attributes, setAttributes, className } ) {
 			) }
 			<InspectorControls group="advanced">
 				<TextControl
-					// TODO: Switch to `true` (40px size) if possible
-					__next40pxDefaultSize={ false }
+					__next40pxDefaultSize
```
Comment (mirka) [inline packages/block-library/src/form-input/edit.js:66]: Enable "Form and input blocks" in Gutenberg Experimental settings. Insert Form block. Click on one of the input blocks.

| Before | After |
|--------|-------|
|FormInput, before|FormInput, after|

File: packages/block-library/src/button/edit.js [hunk 5]
```diff
@@ -380,8 +380,7 @@ function ButtonEdit( props ) {
 			<InspectorControls group="advanced">
 				{ isLinkTag && (
 					<TextControl
-						// TODO: Switch to `true` (40px size) if possible
-						__next40pxDefaultSize={ false }
+						__next40pxDefaultSize
```
Comment (mirka) [inline packages/block-library/src/button/edit.js:383]: | Before | After |
|--------|-------|
|Button, before|Button, after|

File: packages/block-library/src/form/edit.js [hunk 6]
```diff
@@ -130,8 +130,7 @@ const Edit = ( { attributes, setAttributes, clientId } ) => {
 			{ submissionMethod !== 'email' && (
 				<InspectorControls group="advanced">
 					<SelectControl
-						// TODO: Switch to `true` (40px size) if possible
-						__next40pxDefaultSize={ false }
+						__next40pxDefaultSize
```
Comment (mirka) [inline packages/block-library/src/form/edit.js:133]: Enable "Form and input blocks" in Gutenberg Experimental settings. Insert Form block. Choose a custom submission method.

| Before | After |
|--------|-------|
|Form block inspector, before|Form block inspector, after|

File: packages/block-library/src/image/image.js [hunk 7]
```diff
@@ -690,8 +690,7 @@ export default function Image( {
 						) }
 						renderContent={ () => (
 							<TextControl
-								// TODO: Switch to `true` (40px size) if possible
-								__next40pxDefaultSize={ false }
+								__next40pxDefaultSize
```
Comment (mirka) [inline packages/block-library/src/image/image.js:693]: Kind of hard to get to this state. See testing instructions for #58998 if you're interested 😅

| Before | After |
|--------|-------|
|Image block, content only mode, before|Image block, content only mode, after|

File: packages/block-library/src/navigation/edit/navigation-menu-name-control.js [hunk 8]
```diff
@@ -14,8 +14,7 @@ export default function NavigationMenuNameControl() {
 
 	return (
 		<TextControl
-			// TODO: Switch to `true` (40px size) if possible
-			__next40pxDefaultSize={ false }
+			__next40pxDefaultSize
```
Comment (mirka) [inline packages/block-library/src/navigation/edit/navigation-menu-name-control.js:17]: | Before | After |
|--------|-------|
|Navigation block inspector, before|Navigation block inspector, after|

File: packages/block-library/src/post-comment/edit.js [hunk 9]
```diff
@@ -34,8 +34,7 @@ export default function Edit( { attributes: { commentId }, setAttributes } ) {
 					) }
 				>
 					<TextControl
-						// TODO: Switch to `true` (40px size) if possible
-						__next40pxDefaultSize={ false }
+						__next40pxDefaultSize
```
Comment (mirka) [inline packages/block-library/src/post-comment/edit.js:37]: This is a deprecated block so you need to modify the `block.json` to make it appear. (Delete the `__experimental` field, and change `inserter` to `true`.)

| Before | After |
|--------|-------|
|Post Comment block placeholder, before|Post Comment block placeholder, after|

File: packages/block-library/src/post-featured-image/edit.js [hunk 10]
```diff
@@ -230,8 +230,7 @@ export default function PostFeaturedImageEdit( {
 								checked={ linkTarget === '_blank' }
 							/>
 							<TextControl
-								// TODO: Switch to `true` (40px size) if possible
-								__next40pxDefaultSize={ false }
+								__next40pxDefaultSize
```
Comment (mirka) [inline packages/block-library/src/post-featured-image/edit.js:233]: | Before | After |
|--------|-------|
|Featured Image block inspector, before|Featured Image block inspector, after|

File: packages/block-library/src/post-terms/edit.js [hunk 11]
```diff
@@ -83,8 +83,7 @@ export default function PostTermsEdit( {
 			</BlockControls>
 			<InspectorControls group="advanced">
 				<TextControl
-					// TODO: Switch to `true` (40px size) if possible
-					__next40pxDefaultSize={ false }
+					__next40pxDefaultSize
```
Comment (mirka) [inline packages/block-library/src/post-terms/edit.js:86]: Can be seen in the Categories block.

| Before | After |
|--------|-------|
|Category block inspector, before|Category block inspector, after|

File: packages/block-library/src/post-title/edit.js [hunk 12]
```diff
@@ -162,8 +162,7 @@ export default function PostTitleEdit( {
 										checked={ linkTarget === '_blank' }
 									/>
 									<TextControl
-										// TODO: Switch to `true` (40px size) if possible
-										__next40pxDefaultSize={ false }
+										__next40pxDefaultSize
```
Comment (mirka) [inline packages/block-library/src/post-title/edit.js:165]: | Before | After |
|--------|-------|
|Post Title block inspector, before|Post Title block inspector, after|

File: packages/block-library/src/social-link/edit.js [hunk 13]
```diff
@@ -139,8 +139,7 @@ const SocialLinkEdit = ( {
 				<PanelBody title={ __( 'Settings' ) }>
 					<PanelRow>
 						<TextControl
-							// TODO: Switch to `true` (40px size) if possible
-							__next40pxDefaultSize={ false }
+							__next40pxDefaultSize
```
Comment (mirka) [inline packages/block-library/src/social-link/edit.js:142]: Insert "Social Icons" block and add a social.

| Before | After |
|--------|-------|
|Social Icon block inspector, before|Social Icon block inspector, after|

File: packages/block-library/src/template-part/edit/advanced-controls.js [hunk 14]
```diff
@@ -73,8 +73,7 @@ export function TemplatePartAdvancedControls( {
 			{ isEntityAvailable && (
 				<>
 					<TextControl
-						// TODO: Switch to `true` (40px size) if possible
-						__next40pxDefaultSize={ false }
+						__next40pxDefaultSize
```
Comment (mirka) [inline packages/block-library/src/template-part/edit/advanced-controls.js:76]: Can be seen when you select a template part block (like "Header") in the site editor.

| Before | After |
|--------|-------|
|Template Part block inspector, before|Template Part block inspector, after|

File: packages/block-library/src/video/tracks-editor.js [hunk 15]
```diff
@@ -99,8 +99,7 @@ function SingleTrackEditor( { track, onChange, onClose, onRemove } ) {
 				</span>
 				<Grid columns={ 2 } gap={ 4 }>
 					<TextControl
-						// TODO: Switch to `true` (40px size) if possible
-						__next40pxDefaultSize={ false }
+						__next40pxDefaultSize
```
Comment (mirka) [inline packages/block-library/src/video/tracks-editor.js:102]: Can be seen when you add a text track to a Video block. Here is a test file to upload, if you're interested (save with a `.vtt` extension):

```
WEBVTT

00:00:00.500 --> 00:00:02.000
The Web is always changing

00:00:02.500 --> 00:00:04.300
and the way we access it is changing
```

| Before | After |
|--------|-------|
|Video tracks editor, before|Video tracks editor, after|

File: packages/edit-site/src/components/sidebar-dataviews/add-new-view.js [hunk 16]
```diff
@@ -78,8 +78,7 @@ function AddNewItemModalContent( { type, setIsAdding } ) {
 		>
 			<VStack spacing="5">
 				<TextControl
-					// TODO: Switch to `true` (40px size) if possible
-					__next40pxDefaultSize={ false }
+					__next40pxDefaultSize
```
Comment (mirka) [inline packages/edit-site/src/components/sidebar-dataviews/add-new-view.js:81]: Enable "Custom dataviews" in Gutenberg experiments. Go to Pages in the site editor and add a custom view.

| Before | After |
|--------|-------|
|Dataviews Custom view modal, before|Dataviews Custom view modal, after|

File: packages/edit-site/src/components/sidebar-dataviews/custom-dataviews-list.js [hunk 17]
```diff
@@ -51,8 +51,7 @@ function RenameItemModalContent( { dataviewId, currentTitle, setIsRenaming } ) {
 		>
 			<VStack spacing="5">
 				<TextControl
-					// TODO: Switch to `true` (40px size) if possible
-					__next40pxDefaultSize={ false }
+					__next40pxDefaultSize
```
Comment (mirka) [inline packages/edit-site/src/components/sidebar-dataviews/custom-dataviews-list.js:54]: Rename the custom view you just made.

| Before | After |
|--------|-------|
|Rename custom dataview, before|Rename custom dataview, after|

File: packages/editor/src/components/post-publish-panel/postpublish.js [hunk 18]
```diff
@@ -44,7 +44,7 @@ const getFuturePostUrl = ( post ) => {
 function CopyButton( { text, onCopy, children } ) {
 	const ref = useCopyToClipboard( text, onCopy );
 	return (
-		<Button variant="secondary" ref={ ref }>
+		<Button __next40pxDefaultSize variant="secondary" ref={ ref }>
```
Comment (mirka) [inline packages/editor/src/components/post-publish-panel/postpublish.js:47]: The thing that appears after you just publish a post.

| Before | After |
|--------|-------|
|PostPublishPanel, before|PostPublishPanel, after|

File: packages/editor/src/components/post-slug/index.js [hunk 19]
```diff
@@ -24,8 +24,7 @@ function PostSlugControl() {
 
 	return (
 		<TextControl
-			// TODO: Switch to `true` (40px size) if possible
-			__next40pxDefaultSize={ false }
+			__next40pxDefaultSize
```
Comment (mirka) [inline packages/editor/src/components/post-slug/index.js:27]: This doesn't appear anywhere in the app anymore, but code-wise it looks safe to change.

File: packages/editor/src/components/post-taxonomies/hierarchical-term-selector.js [hunk 20]
```diff
@@ -407,8 +407,7 @@ export function HierarchicalTermSelector( { slug } ) {
 		<Flex direction="column" gap="4">
 			{ showFilter && (
 				<TextControl
-					// TODO: Switch to `true` (40px size) if possible
-					__next40pxDefaultSize={ false }
+					__next40pxDefaultSize
```
Comment (mirka) [inline packages/editor/src/components/post-taxonomies/hierarchical-term-selector.js:410]: The category filter that appears in the post sidebar when there are many categories.

| Before | After |
|--------|-------|
|Search categories, before|Search categories, after|


May want to switch to a SearchControl (#64593).

File: packages/reusable-blocks/src/components/reusable-blocks-menu-items/reusable-block-convert-button.js [hunk 21]
```diff
@@ -169,8 +169,7 @@ export default function ReusableBlockConvertButton( {
 					>
 						<VStack spacing="5">
 							<TextControl
-								// TODO: Switch to `true` (40px size) if possible
-								__next40pxDefaultSize={ false }
+								__next40pxDefaultSize
```
Comment (mirka) [inline packages/reusable-blocks/src/components/reusable-blocks-menu-items/reusable-block-convert-button.js:172]: Doesn't appear in the app anymore, but should be safe (same as https://github.com/WordPress/gutenberg/pull/64213#discussion_r1704383798).

File: .eslintrc.js (hunk 1 above)
Comment (ciampo) [reply .eslintrc.js:322]: Neat!

**Review by ciampo — APPROVED** [review]
LGTM 🚀  including the updates to the relevant `Button`s to keep the UI consistent

**Pull request merged**